spec/models/review_spec.rb

```diff
@@ -0,0 +1,73 @@
+require "rails_helper"
+
+RSpec.describe Review, type: :model do
+  let(:user) {FactoryBot.create :user}
+  subject {user}
+
+  let(:tour_detail) {FactoryBot.create :tour_detail}
+  subject {tour_detail}
+
+  let(:review) {FactoryBot.create :review, user: user, 
+                                            tour_detail: tour_detail}
+  subject {review}
+
+  #Test Review model validations
+  context ".validate :" do
```

Comment:
Phân biệt giúp anh context, it, describe

Reply:
> Phân biệt giúp anh context, it, describe

**Context** và **describe**, **context** là một alias method của **describe**. Và hai thằng không có sự khác nhau về chức năng nhưng có khác nhau về ngữ cảnh sử dụng. Như **describe** thì thường để gộp các tests mà test cùng một chức năng, còn **context** thì dùng để gộp các test có cùng chung ngữ cảnh, làm cho test của mình dễ đọc và dễ hiểu hơn.

Còn **it** dùng để biểu thị ví dụ của test, bao gồm 1 đoạn string mô tả và ví dụ. Cũng là nơi thực thi test.